chatterbot/adapters/plugins/evaluate_mathematically.py

```diff
@@ -0,0 +1,200 @@
+from .plugin import PluginAdapter
+import re
+import os, json
+
+class EvaluateMathematically(PluginAdapter):
+
+    def should_answer(self, input_text):
+        """
+        Determines whether it is appropriate for this plugin
+        to respond to the user input.
+        """
+
+        response = self.process( input_text )
+
+        if response == False:
+            return False
+        else:
+            return True
+
+
+    def process(self, input_text):
+        """
+        Takes a statement string.
+        Returns the simplified statement string
+        with the mathematical terms "solved".
+        """
+
+        # Getting the mathematical terms within the input statement
+        expression, string = self.simplify_chunks( self.normalize( input_text ) )
+
+        # Returning important information
+        try:
+            string += '= ' + str( eval( string ) )#self.evaluate( string ) )
+
+            return string
+        except:
+            return False
+
+
+    def simplify_chunks(self, input_text):
+        """
+        Separates the incoming text.
+        """
+
+        expression = []
+        string = ''
+
+        for chunk in input_text.split( ' ' ):
+
+            is_integer = self.isInteger( chunk )
+
+            if is_integer == False:
+                is_float = self.isFloat( chunk )
+
+                if is_float == False:
+                    is_operator = self.isOperator( chunk )
+
+                    if is_operator == False:
+                        continue
+                    else:
+                        expression.append( is_operator )
+
+                        string += str( is_operator ) + ' '
+                else:
+                    expression.append( is_float )
+
+                    string += str( is_float ) + ' '
+            else:
+                expression.append( is_integer )
+
+                string += str( is_integer ) + ' '
+
+        return expression, string
+
+
+    def evaluate( self, expression ):
+        """
+        Evaluates a set of expressions
+        and produces an answer. Then,
+        it returns the answer.
+        """
+
+        return eval( expression )
+
+
+    def isFloat(self, string):
+        """
+        If the string is a float, returns
+        the float of the string. Otherwise,
+        it returns False.
+        """
+
+        try:
+            float( integer )
```

Comment:
It looks like `integer` is undefined here.

Also, this can probably be made to handle the detection of floats without a try, except block:
```python
string.lstrip('-').replace('.','',1).isdigit() # Handles detection of positive and negative floats
```